src/dataprep/main/prep_mag/paper_language.py

```diff
@@ -0,0 +1,194 @@
+            if results == "DONE":
+                break
+            cur.executemany("INSERT INTO paper_language (PaperId, language, score) VALUES (?, ?, ?)", results)
+            con.commit()
+        except Empty:
+            continue
+    con.close()
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Detect language of paper titles")
+    parser.add_argument("--test", action="store_true", help="Run in test mode with 10,000 papers")
+    return parser.parse_args()
+
+def process_chunk(start, end):
+    con = sqlite.connect(database=db_file, isolation_level=None)
+    cur = con.cursor()
+    cur.execute(f"""
+        SELECT PaperId, OriginalTitle
+        FROM Papers
+        LIMIT {end - start} OFFSET {start}
+    """)
+    papers = cur.fetchall()
+    con.close()
+    return process_batch(papers)
+
+def main():
+    args = parse_args()
+    
+    # Connect to MAG db
+    con = sqlite.connect(database=db_file, isolation_level=None)
+    cur = con.cursor()
+
+    # Create paper_language table (drop if exists)
+    cur.execute("DROP TABLE IF EXISTS paper_language")
+    cur.execute("""
+        CREATE TABLE paper_language (
+            PaperId INTEGER PRIMARY KEY,
+            language TEXT,
+            score REAL
+        )
+    """)
+
+    # Get total number of papers
+    if args.test:
+        total_papers = 10000
+    else:
+        cur.execute("SELECT COUNT(*) FROM Papers")
+        total_papers = cur.fetchone()[0]
+
+    # Determine the number of processes to use
+    num_processes = cpu_count()
+
+    # Calculate chunk size based on total papers and number of processes
+    chunk_size = total_papers // num_processes
+    if args.test:
+        chunk_size = min(1000, chunk_size)
+
+    start_time = time.time()
+
+    # Create a manager for shared objects
+    with Manager() as manager:
+        # Create a queue for database operations
+        db_queue = manager.Queue()
+
+        # Start the database worker process
+        db_process = Process(target=db_worker, args=(db_queue, db_file))
+        db_process.start()
+
+        with Pool(processes=num_processes) as pool:
+            # Prepare arguments for each process
+            process_args = []
+            for i in range(num_processes):
+                start = i * chunk_size
+                end = start + chunk_size if i < num_processes - 1 else total_papers
+                process_args.append((start, end))
+
+            # Process papers in parallel
+            results = pool.starmap(process_chunk, process_args)
+
+            # Put results in the queue
+            for batch_result in results:
+                db_queue.put(batch_result)
+
+            # Print progress
+            processed_papers = sum(len(batch) for batch in results)
+            progress = processed_papers / total_papers * 100
+            elapsed_time = time.time() - start_time
+            print(f"Progress: {progress:.2f}% ({processed_papers}/{total_papers}) - "
```

Comment:
I think this prints only at the end of processes, in which I'd suggest to remove it

Reply:
ok!